nodejs/week3/practices/exercise-template/src/backend/api/concerts.js

```diff
@@ -0,0 +1,169 @@
+const express = require("express");
+const { request } = require("../app");
+const router = express.Router();
+const knex = require("../database");
+
+
+router.get("/", async (request, response) => {
+  const concerts = await knex("concerts");
+ 
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+          if(request.query.hasOwnProperty("id")){
+            
+            const requestId=Number(request.query.id);
+            const concertFinder = concerts.some(concert=> concert.id ===requestId);
+            const requestedConcert = await concerts.filter(concert=> concert.id ===requestId);
+    
+            if(concertFinder){
+              response.json(requestedConcert);    
+            }else {
+              response.status(404).send('no concert match your request');
+            }
```

Comment:
It's a little misleading with this response if there is no data that matches the requested id, is that an error?
I guess not, so you can send the same response but without the status(404). 
so you'll have something like  ``response.send('no concert matches your request');``